envelope.go

```diff
@@ -222,6 +222,7 @@ func (w *envelopeWriter) write(env *envelope) *Error {
 type envelopeReader struct {
 	ctx             context.Context //nolint:containedctx
 	reader          io.Reader
+	bytesRead       int64
```

Comment:
Mind adding a comment here explaining what this state is for? We're only using it over in `protocol_grpc.go`, which is pretty far away.